requirements.txt

```diff
@@ -44,4 +45,5 @@ tqdm
 transforms3d
 urllib3[secure]
 pytest_console_scripts
-wquantiles
+voxelmorph
```

Comment:
It looks like installation is currently failing due to the following error:

```
An error occured while importing module voxelmorph -> voxelmorph requires tensorflow version 2.4 or greater, but found version 1.15.5
Full traceback: Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_check_dependencies.py", line 272, in main
    module = module_import(module_name, suppress_stderr)
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_check_dependencies.py", line 104, in module_import
    module = importlib.import_module(module_name)
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/site-packages/voxelmorph/__init__.py", line 52, in <module>
    raise ImportError(f'voxelmorph requires tensorflow version {minv} or greater, '
ImportError: voxelmorph requires tensorflow version 2.4 or greater, but found version 1.15.5
```

The good news is, our dependency on `tensorflow` will be removed shortly. So, once https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738 is merged, we can update this PR against `master`, and the installation errors should be fixed.

Reply:
#3738 has been merged, so I've gone ahead and updated master. We'll see how the affects the installation on this branch. :)

Reply:
Huh. Now we're getting a different error:

```
An error occured while importing module voxelmorph -> Please install tensorflow to use this neurite backend
Full traceback: Traceback (most recent call last):
  File "/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/site-packages/neurite/__init__.py", line 46, in <module>
    import tensorflow
ModuleNotFoundError: No module named 'tensorflow'
```

So, `voxelmorph` is depending on `neurite`, which depends on `tensorflow`. But, we use parts of `voxelmorph` that don't depend on `tensorflow`... 🤔 

That's not good! `voxelmorph` should be installable without needing to have `tensorflow` installed. I'll dig a little deeper into this.

Reply:
The relevant backend-importing code from `neurite` is [here](https://github.com/adalca/neurite/blob/741c95d794f82f91d568eb28ef0c8d1f36d509cd/neurite/__init__.py#L32-L51) with the relevant part being the [`get_backend()`](https://github.com/adalca/neurite/blob/741c95d794f82f91d568eb28ef0c8d1f36d509cd/neurite/py/utils.py#L20) function:

> ```
>     return 'pytorch' if os.environ.get('NEURITE_BACKEND') == 'pytorch' else 'tensorflow'
> ```

Awkward. `neurite` defaults to the `tensorflow` backend, and to prevent it from doing so, we have to manually set an environment variable.

This is doable? But, it would be nice if the upstream dependencies could have handled this a little more gracefully.

Reply:
Perfect! Apart from the linting issues, this PR is ready for testing. :)

Reply:
Perfect ! Thank you very much @joshuacwnewton for having looked into this 👌